### ISSUE TITLE
require("onedarkpro").setup() doesn't seem to have any effect

### ISSUE THREAD

**Issue by anthony-S93** [olimorris/onedarkpro.nvim#83]
Hi @olimorris 

I'm a newbie who recently switched from using vim plug to `packer.nvim`

Following `packer.nvim`'s documentation, the specification I wrote for `onedarkpro.nvim` is something like this:

```lua
local plugins = packer.startup({
    function() --plugin specifications
        local use = packer.use
        use 
        {
            "olimorris/onedarkpro.nvim",
            config = function()
                require("onedarkpro").setup({
                        options = {
                            transparency = true,
                        },
                })
                print('config is ran')
            end
        },
})
return plugins
```
But the setup() function doesn't seem to have any effect at all (the background didn't turn transparent). The print statement at the bottom did run, though. What am I doing wrong? I didn't encounter this issue when I was using vimplug.

**Comment by anthony-S93** [comment]
@olimorris 
**UPDATE:**
I tried writing the config in another file and then `require()` it in my `init.lua`, and this time it worked. Why?

**Comment by olimorris** [comment]
👋🏼 @anthony-S93 and hope you're enjoying the colorscheme.

I have no idea why it hasn't worked. The two methods you've tried above are really the same thing, at least to my knowledge. Are you sure you ran `PackerSync` in your first attempt?

**Comment by anthony-S93** [comment]
@olimorris  

> I have no idea why it hasn't worked. The two methods you've tried above are really the same thing, at least to my knowledge. Are you sure you ran `PackerSync` in your first attempt?

That is strange indeed. That's why I'm so confused. Yes, I did run `PackerSync` the first time.

**Comment by anthony-S93** [comment]
> I just tried copying all of my setup to the `config` function and it worked. I think `PackerStatus` lets you see all of your packer installed plugins so if you try again, you could try running that.

The theme itself was installed. It's just that whatever tweaks I passed to the setup() function have no effect. 
![Screenshot_2022-08-31_21-16-13](https://user-images.githubusercontent.com/69449791/187687379-da7cd887-b7b7-4945-a896-c6a1046ba69d.jpg)

**Comment by olimorris** [comment]
What about if you use the following:

```lua
require("onedarkpro").setup({
  colors = {
    bg = "#FF0000"
  }
})
```

You'll definitely know if it's working then 😄 .

**Comment by anthony-S93** [comment]
> What about if you use the following:
> 
> ```lua
> require("onedarkpro").setup({
>   colors = {
>     bg = "#FF0000"
>   }
> })
> ```
> 
> You'll definitely know if it's working then smile . If you could see the theme's highlights then I'd say it was working just fine. Transparency also depends on your terminal as well as just Neovim.

I added the config above to my packer plugin specification. The background color didn't change. 

That's the thing. I know for certain that transparency works for my terminal. It worked when I wrote `require("onedarkpro").setup()...` in another file and source it in my `init.lua`.

**Comment by anthony-S93** [comment]
![Screenshot_2022-08-31_21-39-57](https://user-images.githubusercontent.com/69449791/187692405-79c2cd04-f12d-4f46-9c55-1cceac975ef0.jpg)

This is after I added the configuration directly into my `init.lua`. The transparency took effect.

**Comment by olimorris** [comment]
Have you added `vim.cmd("colorscheme onedarkpro")` in your plugins file? I notice you didn't do it in your initial post.

**Comment by anthony-S93** [comment]
@olimorris 

I knew it. I knew I messed up somewhere... I added in the vim command inside the function passed to config, and now it's working. 

![Screenshot_2022-08-31_22-04-28](https://user-images.githubusercontent.com/69449791/187698143-7d46b95b-1e09-4868-9eae-7e9d6ee88918.jpg)

My apologies for taking up your time, but can you explain why the command needed to be added into the same function for this to work? I have a pretty good guess why, but I just want to be sure.

**Comment by anthony-S93** [comment]
@olimorris 

When I first made this post, the `vim.cmd("colorscheme onedarkpro")` was inside my `init.lua`, but then all my configs for the theme doesn't take effect. Adding the `vim.cmd("colorscheme onedarkpro")` inside the function fixed the issue. I'm just not sure why it did... (being a noob sucks)

Here's the final version that works for `plugins.lua`:
```lua
local packer = require('packer')

vim.cmd([[
    augroup packer_user_config
        autocmd!
        autocmd BufWritePost plugins.lua source <afile> | PackerCompile
    augroup end
]])

local plugins = packer.startup({
    function() --Plugin specification
        local use = packer.use
        use "wbthomason/packer.nvim"
        use
        {
            "olimorris/onedarkpro.nvim",
            config = function()
                require("onedarkpro").setup({
                    theme = "onedark_vivid",
                    options = {
                        transparency = true,
                        cursorline = true,
                    }
                })
                vim.cmd("colorscheme onedarkpro")
            end,
        }
    end,
    config = { --Packer's config code
        display = {
            open_fn = require('packer.util').float,
        },
    }

})
return plugins
```

**Comment by olimorris** [comment]
If you remove `vim.cmd("colorscheme onedarkpro")` from your config, run `:PackerSync`, reload Neovim, you should see the theme isn't loaded. And if you then run `:lua vim.cmd("colorscheme onedarkpro")` you should see the theme loads. It shouldn't matter where `vim.cmd("colorscheme onedarkpro")` as long as it's after the `onedarkpro.setup` call.

**Comment by anthony-S93** [comment]
@olimorris Indeed, that's what I thought as well. But let me show you the sequence: 

1) I removed `vim.cmd("colorscheme onedarkpro")` and ran `:PackerSync`, and then reloaded Neovim. Like you said, no theme was loaded: 

![Screenshot_2022-08-31_22-28-51](https://user-images.githubusercontent.com/69449791/187703682-4fadfac0-8da0-4061-8577-df66590f1d68.jpg)

2) It shouldn't matter where `onedarkpro.setup` is called as long as it is after the `onedarkpro.setup` call. Which was why I placed the `vim.cmd("colorscheme onedarkpro")` after the `require("plugins")` call in my `init.lua`

![Screenshot_2022-08-31_22-32-32](https://user-images.githubusercontent.com/69449791/187704591-f32f79f8-965a-4b71-9092-282d06b033aa.jpg)

As you can see, the theme did load, but it loaded without the settings I passed into the config function inside `plugins.lua`:

![Screenshot_2022-08-31_22-33-59](https://user-images.githubusercontent.com/69449791/187704882-f6b61d0b-ba66-458c-96fd-98afede4f38f.jpg)

It's almost as if the configs passed into the `onedarkpro.setup` function was ignored? 

3) But now let's say I add `vim.cmd("colorscheme onedarkpro") inside `plugins.lua`, inside the function passed into `config`:

![Screenshot_2022-08-31_22-36-40](https://user-images.githubusercontent.com/69449791/187705499-72d346bb-6569-44bf-bb69-7de174f92ddb.jpg)

As to be expected, this one works. 

Why didn't the first method work? The `vim.cmd("colorscheme onedarkpro")` occurred after the setup function was called also.

**Comment by olimorris** [comment]
I'm out of ideas unfortunately. I just tried the same with mine and it worked as expected:

![Screen Shot 2022-08-31 at 21 44 51](https://user-images.githubusercontent.com/9512444/187778696-bfc5b38d-26b0-46f3-90e6-ab682605edb4.gif)

Some advice, speaking from experience with Neovim config, I wouldn't get too caught up in your config or how you set things up. Packer was a bit of a blackhole for me at the start. I got sucked into the world of lazy loading everything and it just became a nightmare to manage. In my [dotfiles](https://github.com/olimorris/dotfiles), I like to separate out all of my plugins config into individual files so I don't need to run Packer everytime I make a change. I use Packer's `config` method just to call a plugin's file.
